src/components/RegisterForm.test.tsx

```diff
@@ -0,0 +1,205 @@
+
+        const label = getLabelOfInput('name-input')
+        expect(label).toHaveTextContent('Name')
+
+        const input = getFieldOfInput('name-input')
+        expect(input).toHaveValue(exampleUser.name)
+    })
+    test('show firstname input correctly', () => {
+        render(<RegisterForm user={exampleUser} />)
+
+        const label = getLabelOfInput('firstname-input')
+        expect(label).toHaveTextContent('Vorname')
+
+        const input = getFieldOfInput('firstname-input')
+        expect(input).toHaveValue(exampleUser.firstname)
+    })
+    test('show email input correctly', () => {
+        render(<RegisterForm user={exampleUser} />)
+
+        const label = getLabelOfInput('mail-input')
+        expect(label).toHaveTextContent('Mail')
+
+        const input = getFieldOfInput('mail-input')
+        expect(input).toHaveValue(exampleUser.email)
+    })
+    test('show telephone input correctly', () => {
+        render(<RegisterForm user={exampleUser} />)
+
+        const label = getLabelOfInput('telephone-input')
+        expect(label).toHaveTextContent('Telefon')
+
+        const input = getFieldOfInput('telephone-input')
+        expect(input).toHaveValue(exampleUser.telephone)
+    })
+    test('show place input correctly', () => {
+        render(<RegisterForm user={exampleUser} />)
+
+        const label = getLabelOfInput('place-input')
+        expect(label).toHaveTextContent('Ort')
+
+        const input = getFieldOfInput('place-input')
+        expect(input).toHaveValue(exampleUser.place)
+    })
+    test('show street input correctly', () => {
+        render(<RegisterForm user={exampleUser} />)
+
+        const label = getLabelOfInput('street-input')
+        expect(label).toHaveTextContent('Strasse')
+
+        const input = getFieldOfInput('street-input')
+        expect(input).toHaveValue(exampleUser.street)
+    })
+    test('show zip input correctly', () => {
+        render(<RegisterForm user={exampleUser} />)
+
+        const label = getLabelOfInput('zip-input')
+        expect(label).toHaveTextContent('PLZ')
+
+        const input = getFieldOfInput('zip-input')
+        expect(input).toHaveValue(exampleUser.zip)
+    })
+    test('show password input correctly', () => {
+        render(<RegisterForm user={exampleUser} />)
+
+        const label = getLabelOfInput('password-input')
+        expect(label).toHaveTextContent('Passwort')
+
+        const input = getFieldOfInput('password-input')
+        expect(input).toHaveValue(exampleUser.password)
+    })
+    test('show repeat-password input correctly', () => {
+        render(<RegisterForm user={exampleUser} />)
+
+        const label = getLabelOfInput('repeatPassword-input')
+        expect(label).toHaveTextContent('Passwort')
+
+        const input = getFieldOfInput('repeatPassword-input')
+        expect(input).toHaveValue(exampleUser.repeatPassword)
+    })
+
+    test('calls name onChange correctly', () => {
```

Comment:
We could do a similar thing with the `onChange` tests. Specify the configuration in an array: 
```ts
const changeInputs = [
  { testId: 'name-input', newValue: 'new name', changedProperty: 'name' },
  { testId: 'firstname-input', newValue: 'new firstname', changedProperty: 'firstname' },
  ...
]
```

And the test would look like:
```ts
changeInputs.forEach(input => {
  const input = ...
  fireEvent.change(...)

  expect(onChange).toBeCalledTimes(1)
  // Use changedProperty as key:
  expect(onChange).toBeCalledWith({ ...exampleUser, [input.changedProperty]: input.newValue  })

  // probably call this afterwards so the called times statement works
  onChange.mockReset()
})
```

Reply:
improved tests